src/main/java/edu/cornell/kfs/coa/batch/service/impl/DownloadLegacyAccountAttachmentsServiceImpl.java

```diff
@@ -0,0 +1,91 @@
+package edu.cornell.kfs.coa.batch.service.impl;
+
+import java.io.IOException;
+import java.net.URI;
+import java.net.URISyntaxException;
+import java.util.Objects;
+import java.util.regex.Pattern;
+
+import org.apache.commons.lang3.StringUtils;
+import org.apache.commons.lang3.function.FailableBiConsumer;
+import org.apache.hc.core5.net.URIBuilder;
+import org.kuali.kfs.sys.web.WebClientFactory;
+import org.springframework.core.io.buffer.DataBuffer;
+import org.springframework.core.io.buffer.DataBufferUtils;
+import org.springframework.http.HttpStatus;
+import org.springframework.web.reactive.function.BodyExtractors;
+import org.springframework.web.reactive.function.client.ClientResponse;
+import org.springframework.web.reactive.function.client.WebClient;
+
+import edu.cornell.kfs.coa.batch.CuCoaBatchConstants;
+import edu.cornell.kfs.coa.batch.businessobject.LegacyAccountAttachment;
+import edu.cornell.kfs.coa.batch.service.DownloadLegacyAccountAttachmentsService;
+import edu.cornell.kfs.sys.service.WebServiceCredentialService;
+
+public class DownloadLegacyAccountAttachmentsServiceImpl implements DownloadLegacyAccountAttachmentsService {
```

Comment:
I am wondering if re-using AccountingXmlDocumentDownloadAttachmentServiceImpl.downloadByteArray() might simplify this service.

Reply:
That `downloadByteArray()` method was implemented using the JAX-RS client libraries. Because KualiCo is replacing JAX-RS with Spring MVC, I was trying to avoid using JAX-RS here.

I could use Spring's `RestTemplate` instead, but I think Spring is encouraging people to use `WebClient` or `RestClient` instead (where the latter requires Spring 6.x).

Reply:
Upon further analysis, I think I can reuse AccountingXmlDocumentDownloadAttachmentServiceImpl to handle the downloading if I change its primary method to accept more than just `Document` objects. I'll rework things to use this service instead.

Also, we already have KFSPTS-30451 to convert the existing service over to using Spring MVC.